### PR TITLE
Backuptool integration test

### PR DESCRIPTION
**Pull request by swsachith**
This integration test, tests the backup and restore functionality for both the record store and the filesystem.

### CODE REVIEW

File: modules/integration/tests-integration/src/test/java/org/wso2/das/integration/tests/tools/BackupToolTestCase.java [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+            persistedTable = persistenceClient.getAnalyticsTable(BACKUP_TOOL_RECORDSTORE_TEST_STREAM_NAME, STREAM_VERSION);
+        }
+    }
+
+    @Test(groups = "wso2.das.backupTool", description = "Check event stream persistence", dependsOnMethods = "addAnalyticsTable")
+    public void checkDataPersistence() throws Exception {
+        deployEventReceivers();
+        Thread.sleep(20000);
+        publishEvents(1, "Test Event 1", RECORD_COUNT);
+        RecordBean[] records = webServiceClient.getByRange(BACKUP_TOOL_RECORDSTORE_TEST_STREAM_NAME.replace('.', '_'), Long.MIN_VALUE + 1, Long.MAX_VALUE, 0, 1000);
```
Comment (sinthuja) [inline modules/integration/tests-integration/src/test/java/org/wso2/das/integration/tests/tools/BackupToolTestCase.java]: This line seems too big.

File: modules/integration/tests-integration/src/test/java/org/wso2/das/integration/tests/tools/BackupToolTestCase.java [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+        webServiceClient.addStreamDefinition(streamDefinition);
+        AnalyticsTable analyticsTable = getAnalyticsTable();
+        persistenceClient.addAnalyticsTable(analyticsTable);
+
+        int tries = 0;
+        AnalyticsTable persistedTable = persistenceClient.getAnalyticsTable(BACKUP_TOOL_RECORDSTORE_TEST_STREAM_NAME, STREAM_VERSION);
+        while (persistedTable == null) {
+            Thread.sleep(3000);
+            tries++;
+            if(tries == MAX_TRIES)
```
Comment (sinthuja) [inline modules/integration/tests-integration/src/test/java/org/wso2/das/integration/tests/tools/BackupToolTestCase.java]: Include {} for if conditions.

File: modules/integration/tests-integration/src/test/java/org/wso2/das/integration/tests/tools/BackupToolTestCase.java [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+    public void testPurgeData() throws Exception {
+        Date date = new Date();
+        String cronExpressionNow = generateCronExpressionNow(Integer.toString(date.getSeconds()), Integer.toString(date.getMinutes()), Integer.toString(date.getHours()));
+        messageConsoleClient.scheduleDataPurgingTask(BACKUP_TOOL_RECORDSTORE_TEST_STREAM_NAME.replace('.', '_'), cronExpressionNow, -1);
+        RecordBean[] records;
+        int tries = 0;
+        while (true) {
+            Thread.sleep(10000);
+            records = webServiceClient.search(BACKUP_TOOL_RECORDSTORE_TEST_STREAM_NAME.replace('.', '_'), "*:*", 0, RECORD_COUNT * 2);
+            if (records == null || records.length <= 0)
```
Comment (sinthuja) [inline modules/integration/tests-integration/src/test/java/org/wso2/das/integration/tests/tools/BackupToolTestCase.java]: Better to have the if condition with {} for the better readability.

File: modules/integration/tests-integration/src/test/java/org/wso2/das/integration/tests/tools/BackupToolTestCase.java [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+     * @throws Exception
+     */
+    private void publishEvents(long id, String name, int record_count) throws Exception {
+        Event event;
+        dataPublisherClient = new DataPublisherClient();
+        for (int i = 0; i < record_count; i++) {
+            event = new Event(null, System.currentTimeMillis(), new Object[0], new Object[0], new Object[]{id, name});
+            dataPublisherClient.publish(BACKUP_TOOL_RECORDSTORE_TEST_STREAM_NAME, STREAM_VERSION, event);
+        }
+        Thread.sleep(10000);
```
Comment (sinthuja) [inline modules/integration/tests-integration/src/test/java/org/wso2/das/integration/tests/tools/BackupToolTestCase.java]: Remove this thread.sleep in publishing the events.

File: modules/integration/tests-integration/src/test/java/org/wso2/das/integration/tests/tools/BackupToolTestCase.java [hunk 5]
```diff
@@ -0,0 +1,379 @@
...
+    @AfterClass(alwaysRun = true) protected void cleanServer() throws IOException {
+        FileUtils.deleteDirectory(new File(FrameworkPathUtil.getCarbonHome() + File.separator + "test"));
+    }
+
+    @Test(groups = "wso2.das.backupTool", description = "Test backend availability of persistence service") public void testBackendAvailability()
+            throws Exception {
+        Assert.assertTrue(persistenceClient.isBackendServicePresent(), "The persistence service is unavailable");
+    }
+
+    @Test(groups = "wso2.das.backupTool", description = "Adding the new analytics table", dependsOnMethods = "testBackendAvailability") public void addAnalyticsTable()
```
Comment (sinthuja) [inline modules/integration/tests-integration/src/test/java/org/wso2/das/integration/tests/tools/BackupToolTestCase.java]: Code formatting issue.

Comment (swsachith) [inline modules/integration/tests-integration/src/test/java/org/wso2/das/integration/tests/tools/BackupToolTestCase.java]: fixed.

**Pull request closed without merging**